indra/newview/llvoicechannel.cpp

```diff
@@ -122,7 +122,13 @@ void LLVoiceChannel::onChange(EStatusType type, const LLSD& channelInfo, bool pr
 {
     LL_DEBUGS("Voice") << "Incoming channel info: " << channelInfo << LL_ENDL;
     LL_DEBUGS("Voice") << "Current channel info: " << mChannelInfo << LL_ENDL;
-    if (mChannelInfo.isUndefined() || (mChannelInfo.isMap() && mChannelInfo.size() == 0))
+    if (mChannelInfo.has("channel_uri")
```

Comment:
Because you'd also need to compare channel type (webrtc, vivox, or default) and such, this should probably use LLVoiceClient::compareChannels as used below.  You may be able to get away with moving the mChannelInfo assignment below the compareChannels or something.

Reply:
For what cases is that channel assigment needed? Is it webrtc specific?

Reply:
There is a case where you may transition from Vivox to WebRTC when you're in a p2p call, resulting in a different channel uri.  Not quite working yet, but that's the type of stuff I was shooting for.

Reply:
Done. Works for Vivox. Webrtc had no such issue so I'm returning LLSD() there. 

>There is a case where you may transition from Vivox to WebRTC when you're in a p2p call

That seems like it should be calling setChannelInfo, not be assigned on a general listener that processes events from multiple channes.

Milestone was retarretargeted to webrtc, so I rebased to web rtc's branch.